### PR TITLE
Global Beacon interface update

### PR DESCRIPTION
**Pull request by krivi95**
## Description

- Return None by default if dataset doesn't contain Global Beacon:

  Updated `Object.get_global_beacon_status` method so it returns `None` by default if Global Beacon doesn't exist.

  ```python
  dataset = Object.get_by_full_path(dataset_full_path)

  # Returns None
  dataset.get_global_beacon_status()

  # Raises exception: API error 404 {"detail": "No Global Beacon for Dataset:12345679"} 
  dataset.get_global_beacon_status(raise_on_disabled=True)
  ```
- Alias methods in the Dataset class for Global Beacons:
  
  Add ability to perform Global Beacon operations with the `Dataset` class:

  ```python
  dataset = Dataset.get_by_full_path(dataset_full_path)
  dataset.get_global_beacon_status()
  dataset.enable_global_beacon()
  dataset.disable_global_beacon()
  ```
 
- Subject method changes to accommodate API changes:
  
  In `subjects()`/`subjects_count()` methods adding additional parameters in the api request `limit=0` and `include_subjects=True`

### CODE REVIEW

File: solvebio/resource/object.py [hunk 1]
```diff
@@ -621,11 +621,19 @@ def disable_global_beacon(self):
...
+    def get_global_beacon_status(self, raise_on_disabled=False):
         """
         Retrieves the Global Beacon status for this object (datasets only).
         """
         if not self.is_dataset:
             raise SolveError("Only dataset objects can be Global Beacons.")
 
-        return self._client.get(self.instance_url() + '/beacon', {})
+        try:
+            response = self._client.get(self.instance_url() + '/beacon', {})
```
Comment (davecap) [inline solvebio/resource/object.py:632]: To make the code a tiny bit cleaner:

```
try:
    return self._client...
except SolveError:
    if raise_o..:
        raise
    return None
```

**Review by davecap — APPROVED** [review]
Might want to also add those Dataset object aliases (see my comments about the docs)

File: solvebio/global_search.py [hunk 2]
```diff
@@ -185,14 +185,14 @@ def subjects(self):
...
+        self.execute(include_subjects=True, limit=0)
 
         return self._response.get('subjects')
 
     def subjects_count(self):
         """If entity seaarch is performed returns the list of subjects"""
 
         # Executes a query to get a full API response which contains subjects list
-        self.execute()
+        self.execute(include_subjects=True, limit=0)
```
Comment (davecap) [inline solvebio/global_search.py:196]: subjects_count is actually always included, so you can set include_subjects=False here

**Review by davecap — APPROVED** [review]

File: solvebio/global_search.py (hunk 2 above)
Comment (krivi95) [reply solvebio/global_search.py:196]: Maybe I'm wrong but haven't you mentioned on the call that currently subjects are always included, but that we're going to change the API so by default it doesn't return subjects all the time, to make it faster?

Comment (davecap) [reply solvebio/global_search.py:196]: Yes, but the subject count field is always returned regardless.

Comment (krivi95) [reply solvebio/global_search.py:196]: Aha ok ok, understood.

File: solvebio/global_search.py [hunk 3]
```diff
@@ -193,6 +193,6 @@ def subjects_count(self):
         """If entity seaarch is performed returns the list of subjects"""
```
Comment (davecap) [inline solvebio/global_search.py:193]: Docstring is out of date

Comment (krivi95) [reply solvebio/global_search.py:193]: Nice, thanks, missed that one. I'll double-check in the r client as well

**Pull request merged**